rust/datafusion/src/execution/physical_plan/udf.rs

```diff
@@ -44,6 +45,17 @@ pub struct ScalarFunction {
     pub fun: ScalarUdf,
 }
 
+impl Debug for ScalarFunction {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_struct("ScalarFunction")
+            .field("name", &self.name)
+            .field("args", &self.args)
+            .field("return_type", &self.return_type)
+            .field("fun", &"<FUNC>")
```

Comment:
The `ScanarUdf` [link](https://github.com/apache/arrow/pull/7925/files#diff-96f2be3b05d938f6992c8c22d6dd9609L32) is typedef'd to a function pointer and does not implement `Debug` leading to the two implementations in this file